base/applications/osk/main.c

```diff
@@ -457,6 +672,9 @@ BOOL OSK_DlgCommand(WPARAM wCommand, HWND hWndControl)
         PlaySoundW(MAKEINTRESOURCEW(IDI_SOUNDCLICK), GetModuleHandle(NULL), SND_RESOURCE | SND_ASYNC);
     }
 
+    /* FIXME */
+    PostMessageW(Globals.hMainWnd, OSK_KEYPRESS, MapVirtualKeyW(wCommand & SCANCODE_MASK, MAPVK_VSC_TO_VK), 0);
```

Comment:
Fix me but fix me what? Please provide a more meaningful description comment of what should be fixed in the foreseeable future.

Reply:
`Clicking the shift/capslock buttons (not keys) does not update the layout correctly (would appreciate any help with that).`
Clicking buttons does the opposite of what is intended